pkg/cortex/runtime_config.go

```diff
@@ -45,17 +45,38 @@ func loadRuntimeConfig(r io.Reader) (interface{}, error) {
 	return overrides, nil
 }
 
+// tenantLimitsFromRuntimeConfig returns a function that translates tenant IDs to
+// specific limits for that tenant if configured, nil otherwise
 func tenantLimitsFromRuntimeConfig(c *runtimeconfig.Manager) validation.TenantLimits {
 	if c == nil {
 		return nil
 	}
+
+	supplier := tenantLimitsRuntimeConfigFunc(c)
 	return func(userID string) *validation.Limits {
-		cfg, ok := c.GetConfig().(*runtimeConfigValues)
-		if !ok || cfg == nil {
+		tenantLimits := supplier()
+		if tenantLimits == nil {
```

Comment:
Nit: No need for this check... it's actually fine if `tenantLimits` is `nil` here, `tenantLimits[userID]` will then return `nil` too (also fine).